framework/include/utils/DependencyResolver.h

```diff
@@ -226,25 +264,59 @@ DependencyResolver<T>::insertDependency(const T & key, const T & value)
   if (_unique_deps.count(k) > 0)
     return;
   _unique_deps.insert(k);
+  auto insert_it = _depends.insert(k);
 
-  if (dependsOn(value, key))
+  for (auto & pr : _colors)
+    pr.second = WHITE;
+  _colors.emplace(key, WHITE);
+  _colors.emplace(value, WHITE);
+  if (hasCycle(key))
   {
     decltype(_depends) depends_copy(_depends);
-
-    // Insert the breaking dependency here so it will reflect properly in the exception
-    depends_copy.insert(k);
+    _depends.erase(insert_it);
```

Comment:
is this needed? We're throwing anyway

Reply:
Just because you throw doesn't mean the simulation is over. That's the whole reason to throw vs abort